pkg/utils/converter.go

```diff
@@ -473,7 +473,13 @@ func convertStringToTv(schemaType *sdcpb.SchemaLeafType, v string, ts uint64) (*
 	case "boolean":
 		b, err := strconv.ParseBool(v)
 		if err != nil {
-			return nil, err
+			if strings.Contains(strings.ToLower(v), "true") {
```

Comment:
So if the device defines it as boolean but sends a string, then the device implementation is simply wrong. But this check does not do any harm on our side, so thats fine.